### PR TITLE
fix: Default accept nil VE

### PR DESCRIPTION
**Pull request by davidterpay**

### CODE REVIEW

**Comment by aljo242** [comment]
@davidterpay do we need this PR?

**Review by Eric-Warehime — APPROVED** [review]

**Review by aljo242 — APPROVED** [review]

File: abci/ve/vote_extension.go [hunk 1]
```diff
@@ -232,6 +232,17 @@ func (h *VoteExtensionHandler) VerifyVoteExtensionHandler() sdk.VerifyVoteExtens
 			h.logger.Error("VerifyVoteExtensionHandler received a nil request")
 			return nil, err
 		}
+
+		// By default, we accept empty vote extensions.
+		if len(req.VoteExtension) == 0 {
```
Comment (g3ndo9kar9) [inline abci/ve/vote_extension.go:237]: We are we only doing this here, and not replicating in Prepare / Process, etc?

**Comment by g3ndo9kar9** [comment]
If we want to explicitly accept NilVoteExtensions, I'd move this logic to our codecs? That way we aren't duplicating everywhere we have to verify a vote-extension

**Comment by davidterpay** [comment]
> If we want to explicitly accept NilVoteExtensions, I'd move this logic to our codecs? That way we aren't duplicating everywhere we have to verify a vote-extension

@nivasan1 isn't the top level verifyVE the best place to do this? wouldn't we have to replicate this per codec otherwise

**Pull request merged**